### PR TITLE
Feat/support flutter 3.16

### PR DESCRIPTION
**Pull request by DDavidPrime**
#### Overview

> Closes: #626 

Updates to the rx_bloc packages made to support latest flutter version (`3.16`).  Updated dependencies and static analysis warnings/errors in all the packages and example projects (where applicable). Any outstanding changes documented in CHANGELOG files of respective packages.

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected

*Code quality*
- [x] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [x] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations

### CODE REVIEW

File: packages/rx_bloc/pubspec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: rx_bloc
 description: A Flutter package that helps implementing the BLoC (Business Logic Component) Design Pattern using the power of reactive streams.
-version: 5.0.0
+version: 5.1.0
```
Comment (StanevPrime) [inline packages/rx_bloc/pubspec.yaml:3]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/rx_bloc/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## [5.1.0]
```
Comment (StanevPrime) [inline packages/rx_bloc/CHANGELOG.md:1]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/flutter_rx_bloc/pubspec.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: flutter_rx_bloc
 description: Set of Flutter Widgets that help implementing the BLoC design pattern. Built to be used with the rx_bloc package.
-version: 6.1.0
+version: 6.2.0
```
Comment (StanevPrime) [inline packages/flutter_rx_bloc/pubspec.yaml:3]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/flutter_rx_bloc/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+## [6.2.0]
+* Update `rx_bloc` to use version `5.1.0`
```
Comment (StanevPrime) [inline packages/flutter_rx_bloc/CHANGELOG.md:2]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/rx_bloc_list/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## [4.1.0] December 1, 2023
```
Comment (StanevPrime) [inline packages/rx_bloc_list/CHANGELOG.md:1]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/rx_bloc_list/pubspec.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 name: rx_bloc_list
 description: The rx_bloc_list package facilitates implementing infinity scroll and pull-to-refresh features with minimal setup. This package is meant to be used along with RxBloc ecosystem.
-version: 4.0.0
```
Comment (StanevPrime) [inline packages/rx_bloc_list/pubspec.yaml:3]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/rx_bloc_test/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+## [4.1.0] - December 1, 2023
```
Comment (StanevPrime) [inline packages/rx_bloc_test/CHANGELOG.md:1]: When upgrading dev dependencies we do not need to bump the version of the package

File: packages/rx_bloc_test/pubspec.yaml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 name: rx_bloc_test
 description: A Flutter package with the goal to enable testing rxBlocs from the FlutterRxBloc package with ease.
-version: 4.0.0
+version: 4.1.0
```
Comment (StanevPrime) [inline packages/rx_bloc_test/pubspec.yaml:3]: When upgrading dev dependencies we do not need to bump the version of the package

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc/pubspec.yaml (hunk 1 above)
Comment (DDavidPrime) [reply packages/rx_bloc/pubspec.yaml]: Oversight. Noted for next time. ✌🏼

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**